src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1622,10 +1624,12 @@ private IEnumerable<ODataProperty> CreateStructuralPropertyBag(SelectExpandNode
                     }
 
                     ODataProperty property = CreateStructuralProperty(structuralProperty, resourceContext);
-                    if (property != null)
+                    if (property == null || (resourceContext.EdmObject is EdmDeltaDeletedEntityObject && property.Value == null))
```

Comment:
>(resourceContext.EdmObject is EdmDeltaDeletedEntityObject [](http://example.com/codeflow?start=44&length=57)

check this once, before entering loop, rather than for each property.